diff -ruN a/script/framework.py b/script/framework.py
--- a/script/framework.py	2012-05-02 01:44:38.000000000 -0400
+++ b/script/framework.py	2013-01-29 03:15:58.380238199 -0500
@@ -420,12 +420,12 @@
         print 'Checking if mpirun works...',
         sys.stdout.flush()
         # run
-        comm = self.config.mpirun + ' ./tmpc'
-        (output, error, retz) = runShellCommand(comm)
-        if retz:
-            print '\n\nCOMMON: mpirun not working! aborting...'
-            print 'error is:\n','*'*40,'\n',error,'\n','*'*40
-            sys.exit()
+        #comm = self.config.mpirun + ' ./tmpc'
+        #(output, error, retz) = runShellCommand(comm)
+        #if retz:
+        #    print '\n\nCOMMON: mpirun not working! aborting...'
+        #    print 'error is:\n','*'*40,'\n',error,'\n','*'*40
+        #    sys.exit()
 
         # cleanup
         killfiles(['tmpc.c','tmpc'])
@@ -456,21 +456,21 @@
       end\n""")
 
         # compile
-        fcomm = self.config.mpif90+' '+self.config.fcflags+' -o tmpf '+'tmpf.f'
-        (output, error, retz) = runShellCommand(fcomm)
+        #fcomm = self.config.mpif90+' '+self.config.fcflags+' -o tmpf '+'tmpf.f'
+        #(output, error, retz) = runShellCommand(fcomm)
 
-        if retz:
-            print '\n\nCOMMON: mpif90 not working! aborting...'
-            print 'error is:\n','*'*40,'\n',error,'\n','*'*40
-            sys.exit()
+        #if retz:
+        #    print '\n\nCOMMON: mpif90 not working! aborting...'
+        #    print 'error is:\n','*'*40,'\n',error,'\n','*'*40
+        #    sys.exit()
         
         # run
-        comm = self.config.mpirun + ' ./tmpf'
-        (output, error, retz) = runShellCommand(comm)
-        if retz:
-            print '\n\nCOMMON: mpif90 not working! aborting...'
-            print 'error is:\n','*'*40,'\n',error,'\n','*'*40
-            sys.exit()
+        #comm = self.config.mpirun + ' ./tmpf'
+        #(output, error, retz) = runShellCommand(comm)
+        #if retz:
+        #    print '\n\nCOMMON: mpif90 not working! aborting...'
+        #    print 'error is:\n','*'*40,'\n',error,'\n','*'*40
+        #    sys.exit()
 
         # cleanup        
         killfiles(['tmpf.f','tmpf','tmpf.o'])
diff -ruN a/script/utils.py b/script/utils.py
--- a/script/utils.py	2012-05-02 01:44:38.000000000 -0400
+++ b/script/utils.py	2013-01-29 03:16:35.370162963 -0500
@@ -19,7 +19,10 @@
 import re
 import shutil
 
-import popen2
+import warnings
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore",category=DeprecationWarning)
+    import popen2
 
 import select
 
--- a/script/scalapack.py	2014-11-27 07:08:45.072140874 -0500
+++ b/script/scalapack.py	2014-11-27 07:06:51.532142101 -0500
@@ -143,6 +143,16 @@
 
         self.write_slmakeinc()
 
+        print 'Patching pxerbla.f'
+        comm = "sed -i '/WRITE/ s/^/*/' PBLAS/SRC/PTZBLAS/pxerbla.f"
+        sys.stdout.flush()
+        (output, error, retz) = runShellCommand(comm)
+        if retz:
+            print '\n\nScaLAPACK: error patching pxerbla.f'
+            print 'error is:\n','*'*40,'\n',comm,'\n',error,'\n','*'*40
+            writefile(os.path.join(savecwd,'log/scalog'), output+error)
+            sys.exit()
+
         print 'Compiling BLACS, PBLAS and ScaLAPACK...',
         sys.stdout.flush()
         comm = self.make+" lib"
